src/wheel/wheelfile.py

```diff
@@ -60,23 +62,24 @@ def __init__(self, file, mode='r', compression=ZIP_DEFLATED):
                 raise WheelError('Missing {} file'.format(self.record_path))
 
             with record:
-                for line in record:
-                    line = line.decode('utf-8')
-                    path, hash_sum, size = line.rsplit(u',', 2)
-                    if hash_sum:
-                        algorithm, hash_sum = hash_sum.split(u'=')
-                        try:
-                            hashlib.new(algorithm)
-                        except ValueError:
-                            raise WheelError('Unsupported hash algorithm: {}'.format(algorithm))
-
-                        if algorithm.lower() in {'md5', 'sha1'}:
-                            raise WheelError(
-                                'Weak hash algorithm ({}) is not permitted by PEP 427'
-                                .format(algorithm))
-
-                        self._file_hashes[path] = (
-                            algorithm, urlsafe_b64decode(hash_sum.encode('ascii')))
+                for line in csv.reader(TextIOWrapper(record)):
```

Comment:
You need to pass `newline=''` here, see [the docs for the csv module](https://docs.python.org/3/library/csv.html#csv.reader) to make sure you get the correct line endings cross-platform.

(This sort of tricky detail is why hand-rolling a CSV reader like the original code did is a bad idea 😉)

Reply:
Thanks!